File: dmn-core/src/test/resources/tck/cl3/expected/java/dmn/0020-vacation-days/ExtraDaysCase2RuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "ExtraDaysCase2"})
public class ExtraDaysCase2RuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal extraDaysCase2;

    public ExtraDaysCase2RuleOutput(boolean matched) {
        super(matched);
    }

    public java.math.BigDecimal getExtraDaysCase2() {
        return this.extraDaysCase2;
    }
    public void setExtraDaysCase2(java.math.BigDecimal extraDaysCase2) {
        this.extraDaysCase2 = extraDaysCase2;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ExtraDaysCase2RuleOutput other = (ExtraDaysCase2RuleOutput) o;
        if (this.getExtraDaysCase2() != null ? !this.getExtraDaysCase2().equals(other.getExtraDaysCase2()) : other.getExtraDaysCase2() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getExtraDaysCase2() != null ? this.getExtraDaysCase2().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", extraDaysCase2='%s'", extraDaysCase2));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0020-vacation-days/ExtraDaysCase2RuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "ExtraDaysCase2"])
class ExtraDaysCase2RuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("ExtraDaysCase2")
    var extraDaysCase2: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as ExtraDaysCase2RuleOutput
        if (if (this.extraDaysCase2 != null) this.extraDaysCase2 != other.extraDaysCase2 else other.extraDaysCase2 != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.extraDaysCase2 != null) this.extraDaysCase2.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", extraDaysCase2='%s'", extraDaysCase2))
        result_.append(")")
        return result_.toString()
    }
}
